src/bin/pg_autoctl/pgsql.c

```diff
@@ -1310,6 +1320,41 @@ pgsql_get_wal_lag_from_standby(PGSQL *pgsql, int64_t *walLag)
 }
 
 
+/*
+ * pgsql_get_received_lsn_from_standby queries a standby PostgreSQL server to get the
+ * received_lsn value from the pg_stat_wal_receiver system view.
+ *
+ * received_lsn is the latest lsn known to be received and flushed to the disk. It does
+ * not specify if it is applied or not. Caller should have allocated necessary memory
+ * for result value.
+ */
+bool
+pgsql_get_received_lsn_from_standby(PGSQL *pgsql, char *received_lsn)
+{
+	SingleValueResultContext context;
+	char *sql =
+		"SELECT received_lsn"
+		"  FROM pg_stat_wal_receiver";
+
+	context.resultType = PGSQL_RESULT_STRING;
+	context.parsedOk = false;
+
+	log_info("pgsql_get_received_lsn_from_standby : running %s", sql);
+	pgsql_execute_with_params(pgsql, sql, 0, NULL, NULL,
+							  &context, &parseSingleValueResult);
+
+	if (!context.parsedOk)
+	{
+		log_error("PostgreSQL cannot reach the primary server: "
+				  "the system view pg_stat_wal_receiver has no rows.");
+		return false;
+	}
+
+	strlcpy(received_lsn, context.strVal, NAMEDATALEN);
```

Comment:
NAMEDATALEN here should be `size`, as given in argument.

Reply:
done